dkobo/koboform/api.py

```diff
@@ -52,6 +83,30 @@ def delete_survey_draft(self, request, pk=None):
         draft = self.get_object()
         draft.delete()
 
+    def list(self, request, *args, **kwargs):
+        email = request.QUERY_PARAMS.get('email')
+        token = request.QUERY_PARAMS.get('token')
+        username = request.QUERY_PARAMS.get('username')
+        if email and token and username:
+            try:
+                user = User.objects.get(username=username)
+            except User.DoesNotExist:
+                user = User(username=username, password=username)
+                user.email = email
+                user.save()
+
+            payload = {
+                'email': email,
+                'token': token
+            }
+
+            encoded = jwt.encode(
+                payload, settings.JWT_SECRET_KEY, algorithm='HS256')
```

Comment:
algorithm string `HS256` should be replaced everywhere with a configuration variable so that we can alter the algorithm in a uniform way if desired, maybe `JWT_ALGORITHM` is a reasonable name